src/mscorlib/corefx/Interop/Unix/System.Globalization.Native/Interop.Collation.cs

```diff
@@ -25,7 +25,7 @@ internal static partial class GlobalizationInterop
 
         [SecurityCritical]
         [DllImport(Libraries.GlobalizationInterop, CharSet = CharSet.Unicode, EntryPoint = "GlobalizationNative_IndexOf")]
-        internal unsafe static extern int IndexOf(SafeSortHandle sortHandle, string target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options);
+        internal unsafe static extern int IndexOf(SafeSortHandle sortHandle, string target, int cwTargetLength, char* pSource, int cwSourceLength, CompareOptions options, ref int matchLength);
```

Comment:
> ref int matchLength [](start = 170, length = 20)

can we make this parameter as IntPtr? the benefits from that is we can pass IntPtr.Zero when we are not caring to get the matched length and we'll have a better performance.